browser/net/brave_site_hacks_network_delegate_helper_unittest.cc

```diff
@@ -267,4 +268,86 @@ TEST_F(BraveSiteHacksNetworkDelegateHelperTest,
+        context()->CreateRequest(url, net::IDLE, &test_delegate,
+                                 TRAFFIC_ANNOTATION_FOR_TESTS);
+
+    std::shared_ptr<brave::BraveRequestInfo>
+        brave_request_info(new brave::BraveRequestInfo());
+    brave::BraveRequestInfo::FillCTXFromRequest(request.get(),
+        brave_request_info);
+    brave::ResponseCallback callback;
+    int ret = brave::OnBeforeURLRequest_SiteHacksWork(callback,
+        brave_request_info);
+    EXPECT_EQ(ret, net::OK);
+    // new_url should not be set
+    EXPECT_TRUE(brave_request_info->new_url_spec.empty());
+    EXPECT_EQ(request->url(), url);
+  });
+}
+
+TEST_F(BraveSiteHacksNetworkDelegateHelperTest, QueryStringFiltered) {
+  std::vector<std::pair<GURL, GURL> > urls({
+    // { original url, expected url after filtering }
+    { GURL("https://example.com/?fbclid=1234"), GURL("https://example.com/") },
+    { GURL("https://example.com/?fbclid=1234&"), GURL("https://example.com/") },
+    { GURL("https://example.com/?&fbclid=1234"), GURL("https://example.com/") },
+    { GURL("https://example.com/?gclid=1234"), GURL("https://example.com/") },
+    { GURL("https://example.com/?fbclid=0&gclid=1&msclkid=a&mc_eid=a1"),
+      GURL("https://example.com/") },
+    { GURL("https://example.com/?fbclid=&foo=1&bar=2&gclid=abc"),
+      GURL("https://example.com/?fbclid=&foo=1&bar=2") },
+    { GURL("https://example.com/?fbclid=&foo=1&gclid=1234&bar=2"),
+      GURL("https://example.com/?fbclid=&foo=1&bar=2") },
+    { GURL("http://u:p@example.com/path/file.html?foo=1&fbclid=abcd#fragment"),
+      GURL("http://u:p@example.com/path/file.html?foo=1#fragment") },
+    // Obscure edge cases that break most parsers:
+    { GURL("https://example.com/?fbclid&foo&&gclid=2&bar=&%20"),
+      GURL("https://example.com/?fbclid&foo&&bar=&%20") },
+    { GURL("https://example.com/?fbclid=1&1==2&=msclkid&foo=bar&&a=b=c&"),
+      GURL("https://example.com/?1==2&=msclkid&foo=bar&&a=b=c&") },
+    { GURL("https://example.com/?fbclid=1&=2&?foo=yes&bar=2+"),
+      GURL("https://example.com/?=2&?foo=yes&bar=2+") },
+    { GURL("https://example.com/?fbclid=1&a+b+c=some%20thing&1%202=3+4"),
+      GURL("https://example.com/?a+b+c=some%20thing&1%202=3+4") },
+  });
+  std::for_each(urls.begin(), urls.end(), [this](std::pair<GURL, GURL> url){
+    net::TestDelegate test_delegate;
+    std::unique_ptr<net::URLRequest> request =
+        context()->CreateRequest(url.first, net::IDLE, &test_delegate,
+                                 TRAFFIC_ANNOTATION_FOR_TESTS);
+
+    std::shared_ptr<brave::BraveRequestInfo>
+        brave_request_info(new brave::BraveRequestInfo());
+    brave::BraveRequestInfo::FillCTXFromRequest(request.get(),
+        brave_request_info);
+    brave::ResponseCallback callback;
+    int ret = brave::OnBeforeURLRequest_SiteHacksWork(callback,
+        brave_request_info);
+    EXPECT_EQ(ret, net::OK);
+    EXPECT_STREQ(brave_request_info->new_url_spec.c_str(),
```

Comment:
Just `EXPECT_EQ` works perfectly with strings

Reply:
Nice. Thanks for that.